sdk/search/azure-search/src/main/java/com/azure/search/documents/models/PatternAnalyzer.java

```diff
@@ -112,8 +114,9 @@ public RegexFlags getFlags() {
      * @param flags the flags value to set.
      * @return the PatternAnalyzer object itself.
      */
-    public PatternAnalyzer setFlags(RegexFlags flags) {
-        this.flags = flags;
+    public PatternAnalyzer setFlags(Collection<RegexFlags> flags) {
+        String newRegexFlags = flags.stream().map(RegexFlags::toString).collect(Collectors.joining("|"));
```

Comment:
This won't be `null` safe which is different behavior than before.

Reply:
Yes, it will throw NPE. Is it better to make it silent if pass null?

Reply:
`null` was an allowable configuration before, this changes it so that once this is set you can never clear it out.